src/overlays/actors/ovl_En_Rd/z_en_rd.c

```diff
@@ -21,71 +41,869 @@ const ActorInit En_Rd_InitVars = {
+        if (this->unk_31C == 1) {
+            func_8003426C(&this->actor, 0, 0xC8, 0, 0x50);
+        } else {
+            Audio_PlayActorSound2(&this->actor, NA_SE_EN_LIGHT_ARROW_HIT);
+            func_8003426C(&this->actor, -0x8000, 0xC8, 0, 0x50);
+        }
+    }
+    EnRd_SetupAction(this, func_80AE3ECC);
+}
+
+void func_80AE3ECC(EnRd* this, GlobalContext* globalCtx) {
+    if (this->unk_318 != 0) {
+        if (this->unk_316 != 0) {
+            this->unk_316--;
+            if (this->unk_316 >= 0xFF) {
+                func_8003426C(&this->actor, -0x8000, 0xC8, 0, 0xFF);
+            }
+            if (this->unk_316 == 0) {
+                this->unk_318 = 0;
+                gSaveContext.unk_1422 = 0;
+            }
+        }
+    }
+    if (this->actor.dmgEffectTimer == 0) {
+        if (this->actor.colChkInfo.health == 0) {
+            func_80AE2630(globalCtx, &this->actor, 1);
+            func_80AE3C20(this);
+            Item_DropCollectibleRandom(globalCtx, &this->actor, &this->actor.posRot.pos, 0x90);
+        } else {
+            func_80AE3A8C(this);
+        }
+    }
+}
+
+// Regalloc..I can get the score lower by using permuter suggestions, but they all seem silly
+#ifdef NON_MATCHING
+void func_80AE3F9C(EnRd *this, GlobalContext *globalCtx) {
+    s16 unk_310;
+    s16 shapeRotY;
+    s16 yawTowardsLink;
+    s16 new_var;
+    s16 temp_v0;
+    s16 phi_a3;
+    s16 phi_v0;
+
+    unk_310 = this->unk_310;
+    shapeRotY = this->actor.shape.rot.y;
+    yawTowardsLink = this->actor.yawTowardsLink;
+    new_var = unk_310 + shapeRotY;
+    
+    temp_v0 = yawTowardsLink - new_var;
+    phi_a3 = CLAMP(temp_v0, -500, 500);
+
+    temp_v0 -= this->unk_30E;
+    phi_v0 = CLAMP(temp_v0, -500, 500);
+
+    if ((s16)(yawTowardsLink - shapeRotY) >= 0) {
+        this->unk_310 += ABS(phi_a3);
+        this->unk_30E += ABS(phi_v0);
+    } else {
+        this->unk_310 -= ABS(phi_a3);
+        this->unk_30E -= ABS(phi_v0);
+    }
+    
+    this->unk_310 = CLAMP(this->unk_310, -18783, 18783);
+    this->unk_30E = CLAMP(this->unk_30E, -9583, 9583);
+}
+#else
 #pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Rd/func_80AE3F9C.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Rd/func_80AE4114.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Rd/EnRd_Update.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Rd/func_80AE44C4.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Rd/func_80AE4520.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Rd/EnRd_Draw.s")
+#endif
+
+void func_80AE4114(EnRd *this, GlobalContext *globalCtx) {
+    s32 pad;
+    Player *player = PLAYER;
+
+    if ((gSaveContext.unk_1422 != 0) && (this->actor.shape.rot.x == 0) && (this->unk_318 == 0) && (this->unk_31B != 9) && (this->unk_31B != 0xA) && (this->unk_31B != 1)) {
+        func_80AE3DE4(this);
+        return;
+    }
+
+    if (this->collider.base.acFlags & 2) {
```

Comment:
free `else` into the `if` right after, or do you prefer the early return because `func_80AE3DE4` is an action setup?

Reply:
I don't have a strong preference either way. Let's see what others think

Reply:
early return is better here imo